src/app/components/tree/checkbox-tree/terra-checkbox-tree.component.ts

```diff
@@ -49,9 +49,16 @@ export class TerraCheckboxTreeComponent extends TerraBaseTreeComponent implement
      * @param event
      * @param leaf
      */
-    private onCheckboxValueChange(event:boolean, leaf:TerraCheckboxLeafInterface):void
+    protected onCheckboxValueChange(event:boolean, leaf:TerraCheckboxLeafInterface):void
     {
-        leaf.checkboxChecked = event;
+        if(leaf.isIndeterminate)
```

Comment:
Hier wäre es doch besser zu sagen leaf.checkboxChecked = !leaf.isIndeterminate oder?

Reply:
dann kannst du dir das ganze if... xxx =  true else xxx = false sparen

Reply:
ne eben nicht... wenn is indeterminate war, dann soll er es auf false setzen, wenn nicht indeterminate war, dann soll er das vom event nehmen

Reply:
also dann true oder false

Reply:
Oh sorry ^^